File: src/main/java/kr/devport/api/domain/common/config/HikariStartupLogger.java
package kr.devport.api.domain.common.config;

import com.zaxxer.hikari.HikariDataSource;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.stereotype.Component;

import javax.sql.DataSource;
import java.sql.SQLException;

@Component
@RequiredArgsConstructor
@Slf4j
public class HikariStartupLogger implements ApplicationRunner {

    private final DataSource dataSource;

    @Override
    public void run(ApplicationArguments args) {
        HikariDataSource hikariDataSource = resolveHikariDataSource();
        if (hikariDataSource == null) {
            log.warn("Startup datasource log skipped because the active DataSource is not a HikariDataSource: {}", dataSource.getClass().getName());
            return;
        }

        log.info(
            "HikariDataSource startup config resolved: maximumPoolSize={}, minimumIdle={}, connectionTimeout={}ms",
            hikariDataSource.getMaximumPoolSize(),
            hikariDataSource.getMinimumIdle(),
            hikariDataSource.getConnectionTimeout()
        );
    }

    private HikariDataSource resolveHikariDataSource() {
        if (dataSource instanceof HikariDataSource hikariDataSource) {
            return hikariDataSource;
        }

        try {
            if (dataSource.isWrapperFor(HikariDataSource.class)) {
                return dataSource.unwrap(HikariDataSource.class);
            }
        } catch (SQLException e) {
            log.warn("Failed to unwrap DataSource to HikariDataSource for startup logging", e);
        }

        return null;
    }
}

File: application-api/src/main/kotlin/kr/devport/api/domain/common/logging/HikariStartupLogger.kt
package kr.devport.api.domain.common.logging

import com.zaxxer.hikari.HikariDataSource
import org.slf4j.LoggerFactory
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.stereotype.Component
import java.sql.SQLException
import javax.sql.DataSource

@Component
class HikariStartupLogger(
    private val dataSource: DataSource,
) : ApplicationRunner {
    private val log = LoggerFactory.getLogger(javaClass)

    override fun run(args: ApplicationArguments) {
        val hikariDataSource = resolveHikariDataSource()
        if (hikariDataSource == null) {
            log.warn(
                "Startup datasource log skipped because the active DataSource is not a HikariDataSource: {}",
                dataSource.javaClass.name,
            )
            return
        }

        log.info(
            "HikariDataSource startup config resolved: maximumPoolSize={}, minimumIdle={}, connectionTimeout={}ms",
            hikariDataSource.maximumPoolSize,
            hikariDataSource.minimumIdle,
            hikariDataSource.connectionTimeout,
        )
    }

    private fun resolveHikariDataSource(): HikariDataSource? {
        if (dataSource is HikariDataSource) {
            return dataSource
        }

        try {
            if (dataSource.isWrapperFor(HikariDataSource::class.java)) {
                return dataSource.unwrap(HikariDataSource::class.java)
            }
        } catch (e: SQLException) {
            log.warn("Failed to unwrap DataSource to HikariDataSource for startup logging", e)
        }

        return null
    }
}
